lib/npc-generation/setRace.ts

```diff
@@ -6,16 +7,40 @@ import { bmiDescriptions } from './bmiDescriptions'
 import { raceTraits } from './raceTraits'
 import { NPC } from './_common'
 
+export function getGenderTrait (npc: NPC, key: string): unknown {
+  const genderTraits = raceTraits[npc.race].genderTraits
+  // @ts-ignore
+  if (genderTraits[npc.gender][key]) {
+    // @ts-ignore
+    if (typeof genderTraits[npc.gender][key] === 'function') {
+      // @ts-ignore
+      return genderTraits[npc.gender][key]()
+    }
+    // @ts-ignore
+    return genderTraits[npc.gender][key]
+  }
+  // @ts-ignore
+  if (typeof genderTraits.woman[key] === 'function') {
+    // @ts-ignore
+    return genderTraits.woman[key]()
+  }
+  // @ts-ignore
+  return genderTraits.woman[key]
+}
+
 export function setRace (npc: NPC) {
   const raceData = raceTraits[npc.race]
-  const genderTraits = raceData.genderTraits[npc.gender]
-
-  if (random(1, 100) >= genderTraits.beardProbability) {
+  const beardProbability = getGenderTrait(npc, 'beardProbability') as number
+  const baseHeight = getGenderTrait(npc, 'baseHeight') as number
+  const baseWeight = getGenderTrait(npc, 'baseWeight') as number
+  const heightModifier = getGenderTrait(npc, 'heightModifier') as number
+  const weightModifier = getGenderTrait(npc, 'weightModifier') as number
```

Comment:
Seems like `getGenderTraits` would be simpler, as we wouldn't need to use the key as an argument, and could just destructure out the values needed instead of calling the function multiple times.
Example:
```js
const { baseHeight, baseWeight } = getGenderTraits(npc)
```